extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -400,48 +400,55 @@ class CornerstoneViewportService implements IViewportService {
       volume.load();
     });
 
-    this.setVolumesForViewport(viewport, volumeInputArray);
+    return this.setVolumesForViewport(viewport, volumeInputArray);
   }
 
-  public setVolumesForViewport(viewport, volumeInputArray) {
-    viewport.setVolumes(volumeInputArray).then(() => {
-      const viewportInfo = this.getViewportInfo(viewport.id);
-      const initialImageOptions = viewportInfo.getInitialImageOptions();
-
-      if (
-        initialImageOptions &&
-        (initialImageOptions.preset !== undefined ||
-          initialImageOptions.index !== undefined)
-      ) {
-        const { index, preset } = initialImageOptions;
-
-        const { numberOfSlices } = csUtils.getImageSliceDataForVolumeViewport(
-          viewport
-        );
-
-        const imageIndex = this._getInitialImageIndex(
-          numberOfSlices,
-          index,
-          preset
-        );
-
-        csToolsUtils.jumpToSlice(viewport.element, {
-          imageIndex,
-        });
-      }
+  public async setVolumesForViewport(viewport, volumeInputArray) {
+    await viewport.setVolumes(volumeInputArray);
 
-      viewport.render();
-    });
+    const viewportInfo = this.getViewportInfo(viewport.id);
+    const initialImageOptions = viewportInfo.getInitialImageOptions();
+
+    if (
+      initialImageOptions &&
+      (initialImageOptions.preset !== undefined ||
+        initialImageOptions.index !== undefined)
+    ) {
+      const { index, preset } = initialImageOptions;
+
+      const { numberOfSlices } = csUtils.getImageSliceDataForVolumeViewport(
+        viewport
+      );
+
+      const imageIndex = this._getInitialImageIndex(
+        numberOfSlices,
+        index,
+        preset
+      );
+
+      csToolsUtils.jumpToSlice(viewport.element, {
+        imageIndex,
+      });
+    }
+
+    viewport.render();
   }
 
-  public updateViewport(viewportIndex, viewportData) {
+  public updateViewport(viewportIndex, viewportData, keepCamera = false) {
```

Comment:
I really dislike introducing API like this here when the API really isn't general, but should be defined by the hanging protocol and/or external synchronizers - the problem is it adds it as API, which is then hard to change.
For now, could you comment:
@params keepCamera - Temporary API to preserve the camera data.  To be replaced by declared synchronizers/event listeners or other mechanisms.
so that people don't assume this will stay permanently.